department-of-veterans-affairs/abd-vro
# MCP 2316: /v2/health assessment bug

## What was the problem?

If the /vs/health-data-assessment endpoint received a not in scope diagnosis code it would result in a NullPointerException being thrown. 

Associated tickets or Slack threads:

- [MCP-2136](https://amida.atlassian.net/browse/MCP-2136)

## How does this fix it?[^1]

Catch the case that diagnosis lookup fails and reply with a bad request, HTTP status code 400.
## How to test this PR
- Step 1 Try /v2/health-data-assessment with "diagnosticCode":"1111"
- Step 2[^secrel]


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
